core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -378,6 +377,38 @@ private[spark] class SparkSubmit extends Logging {
     val hadoopConf = conf.getOrElse(SparkHadoopUtil.newConfiguration(sparkConf))
     val targetDir = Utils.createTempDir()
 
+    if (clusterManager == EXTERNAL) {
+      val loader = getSubmitClassLoader(sparkConf)
+      sparkConf
+        .get(EXTERNAL_CLUSTER_MANAGER_JARS)
+        .foreach(_
+          .split(",")
+          .map(downloadFile(_, targetDir, sparkConf, hadoopConf))
+          .foreach(addJarToClasspath(_, loader))
+        )
+
+      val maybeSubmitClass = sparkConf.get(EXTERNAL_CLUSTER_SUBMIT_CLASS)
+      if (maybeSubmitClass.isEmpty) {
+        error(
+          "External cluster submit class must be specified when running in external cluster mode."
+        )
+      }
+      maybeSubmitClass.foreach { submitClass =>
+        if (!Utils.classIsLoadable(submitClass) && !Utils.isTesting) {
+          error(
+            s"Could not load external cluster submit class: $submitClass. " +
+              "Make sure the classes are included in the classpath")
+        }
+      }
+      args.maybeMaster match {
+        case Some(url) if SparkContext.getClusterManager(url).isEmpty =>
+          error(s"Master URL is not recognized: $url")
```

Comment:
To reconstruct the old error message as much as possible
```suggestion
          error(s"Master must either be yarn or start with spark, k8s, or local, or be supported by an external cluster manager: $url")
```